packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -772,12 +773,9 @@ export class AskarWallet implements Wallet {
 
           // According to the spec `kid` MUST be a DID URI
           // https://identity.foundation/didcomm-messaging/spec/#construction
-          const recipientDidKey = new DidKey(recipientKey).did
-          const recipientKid = `${recipientDidKey}#${recipientKey.fingerprint}`
+          const recipientKid = new DidKey(recipientKey).didDocument.agreementKeys[0]?.id
```

Comment:
Fixed. Updated wallet pack parameters to accept sender and recipient Did Documents in case of did comm v2. So that we can fetch the correct id of the used key.